experimental/InvariantTheory/src/InvariantTheory.jl

```
@@ -0,0 +1,666 @@
+
+#####################
+#Representation of Reductive Groups. Embeds SLm in GLn. 
+#####################
+
+mutable struct RepresentationReductiveGroup
+    group::ReductiveGroup
+    rep_mat::AbstractAlgebra.Generic.MatSpaceElem
+    sym_deg::Tuple{Bool, Int}
+    reynolds_v::Function
+
+    weights::Vector{Vector{ZZRingElem}}
+    
+    #representation of group G over symmetric degree d
+    function RepresentationReductiveGroup(G::ReductiveGroup, d::Int)
+        R = new()
+        R.group = G
+        R.rep_mat = rep_mat_(G, d)
+        R.sym_deg = (true, d)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+
+    #function RepresentationReductiveGroup(G::ReductiveGroup, W::Union{ZZMatrix, Matrix{<:Integer}, Vector{<:Integer}})
+    function RepresentationReductiveGroup(G::ReductiveGroup, W::Vector{Vector{ZZRingElem}})
+        if !(G.group[1] == :torus)
+            return nothing
+        else
+            R = new()
+            R.group = G
+            R.weights = W
+            return R
+        end
+    end
+    
+    #matrix M is the representation matrix. does not check M.
+    function RepresentationReductiveGroup(G::ReductiveGroup, M::AbstractAlgebra.Generic.MatSpaceElem)
+        @req base_ring(M) == base_ring(G.group_ideal) "Group ideal and representation matrix must have same parent ring"
+        R = new()
+        R.group = G
+        R.rep_mat = M
+        R.sym_deg = (false, 0)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+end
+
+#representation_reductive_group(G::ReductiveGroup, d::Int) = RepresentationReductiveGroup(G, d)
+representation_reductive_group(G::ReductiveGroup, M::AbstractAlgebra.Generic.MatSpaceElem) = RepresentationReductiveGroup(G,M)
+group(R::RepresentationReductiveGroup) = R.group
+
+function representation_on_forms(G::ReductiveGroup, d::Int)
+    if G.group[1] == :SL
+        return RepresentationReductiveGroup(G, d)
+    end
+end
```

Shouldn't this function throw an exception instead of silently returning `nothing`?